### PR TITLE
[sysdig-deploy][admission-controller][agent][kspm-collector][node-ana…

### PR DESCRIPTION
**Pull request by yashgiri**
…lyzer][rapid-response] Added golden unit tests

## What this PR does / why we need it:

Golden templates - Pre-generated set of templates for a certain set of defined values.
generate golden template unit tests (most common and best-known edge cases)
## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with chart name (e.g. [mychartname])
- [x] Chart Version bumped for the respective charts
- [x] Changelog updated for the charts with version updates.
- [ ] Variables are documented in the README.md (or README.tpl in some charts)
- [ ] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

File: charts/admission-controller/tests/golden_template_test.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    documentIndex: 0
+    set:
+      sysdig:
+        secureAPIToken: standard_token
+      global:
+        sysdig:
+          region: us2
+      clusterName: test-k8s
+      scanner:
+        enabled: true
```
Comment (ironashram) [inline charts/admission-controller/tests/golden_template_test.yaml]: this defaults to true, is it necessary to add it here?

File: charts/admission-controller/tests/golden_template_test.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      webhook:
+        enabled: true
```
Comment (ironashram) [inline charts/admission-controller/tests/golden_template_test.yaml]: i don't see `webhook.enabled `  key in the values at all

**Review by ironashram — CHANGES_REQUESTED** [review]
left two comments regarding the admission-controller golden test, apart from that lgtm

File: charts/admission-controller/tests/golden_template_test.yaml (hunk 1 above)
Comment (yashgiri) [reply charts/admission-controller/tests/golden_template_test.yaml]: Got it I think I will remove this then.

File: charts/admission-controller/tests/golden_template_test.yaml (hunk 2 above)
Comment (yashgiri) [reply charts/admission-controller/tests/golden_template_test.yaml]: Removed it. I must have assumed there was while running it good to know.

**Review by ironashram — APPROVED** [review]
